packages/c/sshnpd/src/main.c

```diff
@@ -256,78 +270,68 @@ int main(int argc, char **argv) {
 
   if (ping_response == NULL) {
     atlogger_log(LOGGER_TAG, ATLOGGER_LOGGING_LEVEL_ERROR, "cJSON_Print failed\n");
-    goto cancel_atclient;
-  } else {
-    free_ping_response = true;
+    atcommons_memlist_failure_free(&memlist);
+    return res;
+  }
+  res = atcommons_memlist_add(&memlist, ping_response, true, NULL);
+  if (res != 0) {
+    free(ping_response);
+    atcommons_memlist_failure_free(&memlist);
+    return res;
   }
-
   if (!should_run) {
-    goto cancel_atclient;
+    atcommons_memlist_failure_free(&memlist);
+    return res;
   }
 
   // 9. Start the device refresh loop - if hide is off
-  pthread_t refresh_tid;
-  atclient_atkey *infokeys = malloc(sizeof(atclient_atkey) * params.manager_list_len);
-  if (infokeys == NULL) {
-    atlogger_log(LOGGER_TAG, ATLOGGER_LOGGING_LEVEL_ERROR, "Failed to allocate memory for infokeys\n");
-    exit_res = 1;
-    goto cancel_atclient;
-  }
-
-  for (size_t i = 0; i < params.manager_list_len; i++) {
-    atclient_atkey_init(infokeys + i);
-  }
-
-  atclient_atkey *usernamekeys = malloc(sizeof(atclient_atkey) * params.manager_list_len);
-  if (usernamekeys == NULL) {
-    atlogger_log(LOGGER_TAG, ATLOGGER_LOGGING_LEVEL_ERROR, "Failed to allocate memory for usernamekeys\n");
-    exit_res = 1;
-    goto clean_info_keys;
-  }
-
-  for (size_t i = 0; i < params.manager_list_len; i++) {
-    atclient_atkey_init(usernamekeys + i);
-  }
-
-  struct refresh_device_entry_params refresh_params = {
-      &worker, &atclient_lock, &refresh_cond, &params, ping_response, username, &should_run, infokeys, usernamekeys};
-  res = pthread_create(&refresh_tid, NULL, refresh_device_entry, (void *)&refresh_params);
+  res = handle_username_keys(&worker, (const char **)params.manager_list, params.manager_list_len, username,
```

Comment:
Instead of starting the background thread, we only need to handle_username_keys during initial startup now.